### DISCUSSION TITLE
Ingredient and Nutrition Info should have another parent

### DISCUSSION THREAD

**Discussion by markwbrown** [schemaorg/schemaorg#3039] [Ideas]
Currently, Ingredients exist in Recipe which, along with MenuItem, also contains NutritionInformation. 

Every time you're at the grocery store, you walk by aisles of products containing NutritionInformation and Ingredients. But these items aren't Recipes, per se. And they aren't really MenuItems either. I think Product may benefit from adding these property fields to the model. Anybody else?

**Comment by thadguidry** [comment]
You are right that NutritionInformation is missing on Product and it's at the SKU level and we have sku property on Product.  There was just bad timing on the release of NutritionInformation I would say and so was missed on Product.  YOU could internally begin adding markup with the nutrition property on Product and expect that to eventually be accepted in validators and a future release of Schema.org.  My validator accepts it already on Product and fails if found on IndividualProduct (serialized items).

Besides directly with Schema.org, you can certainly use the GS1 extension and making a nutritional claim statement with gs1:nutritionalInfo.  Your best bet is to use JSON-LD format in this case.  Take a look online at markup examples on various previous blog posts and developer [repos](https://github.com/gs1/WebVoc) that show examples of combining Schema.org with GS1.

@philarcher any other thoughts here?

**Comment by gmboxsmg2psn-collab** [comment]
Согласен, логично звучит. Сейчас получается, что NutritionInformation и Ingredients привязаны к Recipe/MenuItem, хотя в реальности это более универсальные сущности. Добавление отдельной модели Product с этими полями сделает структуру гибче и ближе к реальному миру — продукты из магазина действительно не являются рецептами, но при этом содержат ту же информацию.

Такой подход упростит переиспользование данных и снизит дублирование, особенно если один и тот же продукт может входить в разные рецепты. В целом идея выглядит здравой и масштабируемой — всё зависит от того, насколько глубоко вы хотите развивать модель. Иногда архитектура тоже напоминает выбор решений в разных сценариях, как и в [Win casino автоматы онлайн](https://win.casino/), где важно подобрать оптимальный вариант под задачу

**Reply by markwbrown** [reply]
what in the AI slop is this garbage?
